modules/_labs/button/react/lib/Button.tsx

```diff
@@ -0,0 +1,227 @@
+import * as React from 'react';
+import {colors} from '@workday/canvas-kit-react-core';
+import {CanvasSystemIcon} from '@workday/design-assets-types';
+import {GrowthBehavior} from '@workday/canvas-kit-react-common';
+import {ButtonVariant, ButtonSize, ButtonColorCollection} from './types';
+import {ButtonContainer, ButtonLabel, ButtonLabelData, ButtonLabelIcon} from './parts';
+
+export interface ButtonProps extends React.HTMLAttributes<HTMLButtonElement>, GrowthBehavior {
+  /**
+   * The variant of the Button.
+   * @default ButtonVariant.Secondary
+   */
+  variant: ButtonVariant;
+  /**
+   * The size of the Button.
+   * @default ButtonSize.Medium
+   */
+  size: ButtonSize;
+  /**
+   * The ref to the button that the styled component renders.
+   */
+  buttonRef?: React.Ref<HTMLButtonElement>;
+  /**
+   * The data label of the Button.
+   */
+  dataLabel?: String;
+  /**
+   * The icon of the Button.
+   */
+  icon?: CanvasSystemIcon;
+}
+
+const Button = (props: ButtonProps) => {
+  const {variant, size, buttonRef, dataLabel, icon, children, ...elemProps} = props;
+
+  return (
+    // eslint-disable-next-line no-use-before-define
+    <ButtonContainer colors={buttonColors[variant]} size={size} ref={buttonRef} {...elemProps}>
+      {icon && <ButtonLabelIcon size={size} icon={icon} />}
+      <ButtonLabel size={size} variant={variant}>
+        {children}
+      </ButtonLabel>
+      {dataLabel && <ButtonLabelData size={size}>{dataLabel}</ButtonLabelData>}
+    </ButtonContainer>
+  );
+};
+
+Button.Variant = ButtonVariant;
+Button.Size = ButtonSize;
+
+Button.defaultProps = {
+  variant: ButtonVariant.Secondary,
+  size: ButtonSize.Medium,
+};
```

Comment:
What is the difference between using `defaultProps` and using defaults in the functional component?

```tsx
export interface ButtonProps extends React.HTMLAttributes<HTMLButtonElement>, GrowthBehavior {
  /**
   * The variant of the Button.
   * @default ButtonVariant.Secondary
   */
  variant?: ButtonVariant;
  /**
   * The size of the Button.
   * @default ButtonSize.Medium
   */
  size?: ButtonSize;
  // ...
}

const {
  variant = ButtonVariant.Secondary,
  size = ButtonSize.Medium,
  buttonRef,
  dataLabel,
  icon,
  children,
  ...elemProps
} = props;
```

This is less type magic. It does have the downside of evaluating the right-hand for every render cycle (no difference in this case). I don't have a strong preference either way. I've exclusively done default assignment in the past. I'm fine with this if Typescript supports it well (makes `variant` and `size` optional externally) and documentation generators understand it.

Reply:
I would be fine with this approach. Only reason I did it the other way was because that's how I've seen it done in the majority of examples I've looked at.

Reply:
@NicholasBoll Update: Typescript doesn't seem to recognize these as default props when you do this. I get an error wherever I don't specify a variant and size after this change